java/cdk-environment/src/test/java/sleeper/environment/cdk/buildec2/KeyPairUtilTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KeyPairUtilTest {
+
+    @Test
+    public void canGetPublicKeyInBase64() {
+        assertThat(KeyPairUtil.publicBase64(KeyPairUtil.generate()))
```

Comment:
It's not ideal to use a method in the class under test to generate the key-pair for the purposes of testing a different method - could we create a test key-pair and put it in a file under test/resources?

Reply:
Then we'd need to be able to read it from disk as well. I started implementing that but decided it wasn't worth it.

Reply:
A more pressing issue is that there's no guarantee that reading the key-pair from disk would be representative of the object you'd get from generating a fresh one.

Reply:
That's where you'd need another test to test the generate method, which is something you're currently lacking.

Reply:
I've done that, thanks.